rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java

```diff
@@ -242,12 +242,18 @@ public URI buildEncodedUri(String endpointUrl) {
         return uri;
     }
 
-    private String getData(TbMsg msg) {
-        String data = msg.getData();
+    private String getData(TbMsg tbMsg) {
+        return trimDoubleQuotesIfRequired(tbMsg.getData());
+    }
 
-        if (config.isTrimDoubleQuotes()) {
+    protected String trimDoubleQuotesIfRequired(String data) {
```

Comment:
1. The feature has to be renamed to make more sense. as it acts as JSON.parse to make plain/text from JSON string.
2. UI and description and documentation have to be adjusted.
3. the method getData should get `getConfig().isTrimDoubleQuotes()`as a boolean parameter. This make your code better and much easier to test without excessive mocking